### PR TITLE
[cli][client][tests] update to node14-compatible target (drop support for node12)

### PR DESCRIPTION
**Pull request by EndangeredMassa**
Node 12 is EOL and we're starting to reference packages that don't install on node lower than 14, such as nuxt@3 and remix. Let's update to 14 for now.

---

Reimplements: https://github.com/vercel/vercel/pull/7819

### CODE REVIEW

**Review by styfle — APPROVED** [review]
Thanks!

**Comment by TooTallNate** [comment]
Unfortunately Jest is doing something different with the dynamic requires that makes the Unit tests not fail here, but with the built CLI I am getting this with `vc build`:

```
> [debug] [2022-05-25T19:00:07.084Z] Failed to import "@vercel/static-build": Error: Cannot find module '@vercel/static-build/package.json'
> [debug] [2022-05-25T19:00:07.085Z] Installing Builders: @vercel/static-build
yarn add v1.22.18
warning ../../../package.json: No license field
info No lockfile found.
[1/4] 🔍  Resolving packages...
...
Error! Importing "@vercel/static-build": Cannot find module '/Users/nrajlich/r/.vercel/builders/node_modules/@vercel/static-build/dist/index'
```

**Comment by TooTallNate** [comment]
We probably need to do the `eval('require')` stuff in `import-builders.ts` as well.

File: packages/next/test/fixtures/00-i18n-support/index.test.js
Comment (styfle) [inline packages/next/test/fixtures/00-i18n-support/index.test.js]: This might be worth moving to a separate PR: seems like a bug when the tests were written?

File: packages/next/src/templated-launcher.ts
Comment (TooTallNate) [inline packages/next/src/templated-launcher.ts]: Are we sure this one is necessary? I don't believe this file is compiled by webpack.

Comment (TooTallNate) [reply packages/next/src/templated-launcher.ts]: @ijjk could maybe confirm.

File: packages/next/src/utils.ts
Comment (TooTallNate) [inline packages/next/src/utils.ts]: I'd suggest `fs.readJSON()` instead.

File: packages/next/src/templated-launcher.ts
Comment (ijjk) [reply packages/next/src/templated-launcher.ts]: Yeah this one shouldn't be touched by `ncc` just `tsc` x-ref: https://github.com/vercel/vercel/blob/0b56caba45c4ab0a74e7ae0a451cda033db2db04/packages/next/build.js#L9

**Comment by TooTallNate** [comment]
I pushed a failing CLI test for `vc build` (from #7876) to this branch to make sure that the test is passing before we merge this.

File: packages/next/test/fixtures/00-i18n-support/index.test.js
Comment (TooTallNate) [reply packages/next/test/fixtures/00-i18n-support/index.test.js]: It's not a bug, it's just that the first test creates the deployment and the other tests perform some assertions on the behavior of the deployment.

This `index.test.js` file is afaik copy/pasted into all the other `next/test/fixtures` directories so if we're making a change here then it should probably be applied to all of them.

**Comment by EndangeredMassa** [comment]
Holding off on merging until we get a new minor release out. Then this can be in a new major release.

File: packages/next/test/fixtures/00-i18n-support/index.test.js
Comment (EndangeredMassa) [reply packages/next/test/fixtures/00-i18n-support/index.test.js]: I would consider this a bug because it relies on test order to execute properly.

I reverted the change here and will make a change in all of these files in a separate PR.

**Review by AndyBitz — APPROVED** [review]

File: packages/next/src/templated-launcher.ts
Comment (styfle) [inline packages/next/src/templated-launcher.ts]: Should this one use `eval()` too? Why does lint no longer complain?

File: packages/next/src/utils.ts
Comment (styfle) [inline packages/next/src/utils.ts]: Lets use `fs.readJSON()` like we do for routes manifest above

File: packages/next/src/utils.ts
Comment (styfle) [inline packages/next/src/utils.ts]: Lets use `fs.readJSON()` like we do for routes manifest above

File: packages/node-bridge/tsconfig.json
Comment (styfle) [inline packages/node-bridge/tsconfig.json]: I'm not sure this is correct. The `module` should likely be `commonjs` since we even changed TS files to JS in this package to preserve CJS

File: packages/next/src/templated-launcher.ts
Comment (EndangeredMassa) [reply packages/next/src/templated-launcher.ts]: This one doesn't need to be `eval`, according to: https://github.com/vercel/vercel/pull/7865#discussion_r882232084

The linting rule is off for some reason. I enabled it and restored this opt-out line. [`a3c5a3d` (#7865)](https://github.com/vercel/vercel/pull/7865/commits/a3c5a3d37a2474d97fddcd500bb513721de32b26)

Comment (EndangeredMassa) [reply packages/next/src/templated-launcher.ts]: Ahh, that's why the rule was off: https://github.com/vercel/vercel/runs/6698194648?check_suite_focus=true

I'll mess with that in a separate PR. Turning this rule back off.

File: packages/node-bridge/tsconfig.json
Comment (EndangeredMassa) [reply packages/node-bridge/tsconfig.json]: Makes sense. I updated it to `commonjs`. [`d905a75` (#7865)](https://github.com/vercel/vercel/pull/7865/commits/d905a759cffded27b6cb4ce8a2ec74a3176284b5)

File: packages/next/src/templated-launcher.ts
Comment (EndangeredMassa) [reply packages/next/src/templated-launcher.ts]: There are 247 violations of this rule. We can discuss if we want to enforce it or not. Maybe for now we remove these comments disabling the rule on a per-line level.

**Review by TooTallNate — APPROVED** [review]

**Pull request merged**
